uniffi/src/lib.rs

```diff
@@ -23,212 +45,291 @@ pub mod deps {
     pub use log;
 }
 
-/// Any type that can be returned over the FFI must implement the `Lowerable` trait, to define how
-/// it gets lowered into bytes for transit. We provide default implementtions for primitive types,
-/// and a typical implementation for composite types would lower each member in turn.
+/// Trait defining how to transfer values via the FFI layer.
+///
+/// The `ViaFfi` trait defines how to pass values of a particular type back-and-forth over
+/// the uniffi generated FFI layer, both as standalone argument or return values, and as
+/// part of serialized compound data structures.
+///
+/// (This trait is Like the `InfoFfi` trait from `ffi_support`, but local to this crate
+/// so that we can add some alternative implementations for different builtin types,
+/// and so that we can add support for receiving as well as returning).
+///
+/// ## Safety
+///
+/// This is an unsafe trait (implementing it requires `unsafe impl`) because we can't guarantee
+/// that it's safe to pass your type out to foreign-language code and back again. Buggy
+/// implementations of this trait might violate some assumptions made by the generated code,
+/// or might not match with the corresponding code in the generated foreign-language bindings.
+///
+/// In general, you should not need to implement this trait by hand, and should instead rely on
+/// implementations generated from your component IDL via the `uniffi-bindgen scaffolding` command.
+
+pub unsafe trait ViaFfi: Sized {
```

Comment:
I like that this trait will ensure we don't miss Lowerable/Liftable impls anymore since everything is in a single trait now :100: